File: fsharp/rna-transcription/RNATranscription.fs
module RNATranscription

let chartToRNA char =
    match char with
    | 'G' -> 'C'
    | 'C' -> 'G'
    | 'T' -> 'A'
    | 'A' -> 'U'
    | _ -> char

let toRna seq =
    seq |> String.map chartToRNA

File: ocaml/rna-transcription/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let to_rna =
    List.map (fun c ->
        match c with
        | `G -> `C
        | `C -> `G
        | `T -> `A
        | `A -> `U
    )